subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -350,6 +356,56 @@ static int tls_release(struct tls_context *tls)
 	return 0;
 }
 
+#if defined(CONFIG_NET_SOCKETS_ENABLE_DTLS)
+static bool dtls_is_peer_addr_valid(struct net_context *context,
+				    const struct sockaddr *peer_addr,
+				    socklen_t addrlen)
+{
+	if (context->tls->dtls_peer_addrlen != addrlen ||
+	    context->tls->dtls_peer_addr.sa_family != peer_addr->sa_family) {
+		return false;
+	}
+
+	if (peer_addr->sa_family == AF_INET6) {
```

Comment:
We could optimize this a bit more, so something like this

`if (IS_ENABLED(CONFIG_NET_IPV6) && peer_addr->sa_family == AF_INET6) {`